### PR TITLE
Assignment/1

### PR DESCRIPTION
**Pull request by Ranapriyanshi**
🔅Submission for Assignment 1 - **BuzzOn**

🌀**DBMS concepts  - - [✔️] ERD and Schema** 

_ERD for Buzzon -_ 
image

_Schema for Buzzon -_ 
image

Here is also the drive link for these files - [BuzzOn ERD and Schema Images](https://drive.google.com/drive/folders/1xs1TD0SXGNN1-4BmWqNLZYqppsJ13lEJ)

🌀**OOPS concepts  -**  
 - [✔️] **Classes and Objects**     
 - [✔️ ] **`This ` Pointer**

Description - Created different classes and objects per the application's requirements.

### CODE REVIEW

**Comment by PRASANNAPATHI** [comment]
Add the loom video explaining the diagrams' connections and working.

**Comment by th3c0d3br34ker** [comment]
Kindly add loom video explaining Classes and Objects as well.

File: server/models/verifiedUser.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+export class
+VerifiedUser{
```
Comment (th3c0d3br34ker) [inline server/models/verifiedUser.ts:2]: consider: `VerifiedUser` can extend the base `User` class.

**Review by th3c0d3br34ker — COMMENTED** [review]
thanks @Ranapriyanshi, i have reviewed your submission. 

Overall comments:

1. Please demonstrate the creation of objects.

Please add a loom demonstrating the concepts for Classes and Objects and let me know.

> Note: Adding a loom if mandatory for the review.

**Comment by Ranapriyanshi** [comment]
Hey @PRASANNAPATHI, here are the loom video recordings explaining  ERD and Schema for BuzzOn individually.
✨[Loom Video explaining ERD.](https://www.loom.com/share/bbbe1ad99b124858b7edeb10b2c15a66?sid=4da7a5f7-4d8d-4960-9219-e2c6b01d60b9)
✨[Loom Video explaining BuzzOn Schema](https://www.loom.com/share/0bbbb6e8c7394551b681146853bc522b?sid=92113913-2779-4f6b-865c-7b0222f63786)

File: server/models/verifiedUser.ts (hunk 1 above)
Comment (Ranapriyanshi) [reply server/models/verifiedUser.ts:2]: Updated `verifiedUser` to be an extension to the base class `User`

**Comment by Ranapriyanshi** [comment]
Hey @th3c0d3br34ker, Thanks for the review.
✨Here is the loom video explaining [Classes and Object and `This` pointer.](https://www.loom.com/share/7405f224678c40b489e1cd8b39337fc4?sid=0d139fe0-5f79-42bc-a430-ccfa03461ea1)

Here, are screenshots demonstrating Objects as well.
image
image

**Comment by th3c0d3br34ker** [comment]
hey @Ranapriyanshi, please schedule a catchup.

Take note:
1. Keep the project ready on your local when joining the meet.
2. The catchup will address both DBMS and OOP submission.

**Review by th3c0d3br34ker — APPROVED** [review]
thanks @Ranapriyanshi for your submission,

For DBMS (`ERD and Schema` Concept):

1. consider combining the `User` and `Verified User` entity, by adding additional attributes in the `User` table.
2. the `cardinality and ordinality` for the relationships are missing in the ERD.

Please consider the above points for DBMS evaluation.

For OOP (`Classes and Objects `and `this` pointer Concepts)

1. The self-viva is sufficient for the mentioned concepts. 
2. a demo is missing in the self-viva.

Consider raising separate PRs for each concept for DBMS and OOP.

> Please raise individual PRs going forward. 

As per our viva discussion. You may merge this PR.

**Comment by Ranapriyanshi** [comment]
Hey @th3c0d3br34ker, Thanks for the review! I'll take note of above-mentioned point.😊

**Pull request merged**